stumpy/aampdist_snippets.py

```diff
@@ -8,6 +8,30 @@
 from .aampdist import _aampdist_vect
 
 
+def _get_mask_slices(idx, mask):
+    """
+    For a boolean vector mask, returns the slices of indices at which the mask is True.
+
+    Parameters
+    ----------
+    mask: boolean ndarray
+
+    Returns
+    -------
+    slices: slices of indices where the mask is True
+
+    """
+    m1 = np.r_[0, mask]
+    m2 = np.r_[mask, 0]
+
+    (starts, ) = np.where(~m1 & m2)
+    (ends, ) = np.where(m1 & ~m2)
+
+    slices = np.c_[[idx] * len(starts), starts, ends]
```

Comment:
Can you please explain what is happening here in this line? The original code had:

```
def _get_mask_slices(mask):
    m1 = np.r_[0, mask]
    m2 = np.r_[mask, 0]
    starts, = np.where(~m1 & m2)
    ends, = np.where(m1 & ~m2)
    return np.c_[starts, ends]
```

and this looks different

Reply:
I wanted to provide the snippet ID as well when I return the slices. (However, as you said, it is better to use the reusable version and change the output to include the snippet ID).

Reply:
Ahhh, now I understand. Thank you for clarifying it for me!